Keen/KeenClient.cs

```diff
@@ -970,5 +973,177 @@ public JObject Query(string queryName, Dictionary<string, string> parms)
                 throw ex.TryUnwrap();
             }
         }
+
+        /// <summary>
+        /// Get query results from a Cached Dataset
+        /// </summary>
+        /// <param name="datasetName">Name of cached dataset to query.</param>
+        /// <param name="indexBy">The string property value you want to retrieve results by.</param>
+        /// <param name="timeframe">Limits retrieval of results to a specific portion of the Cached Dataset</param>
+        /// <returns>An instance of Newtonsoft.Json.Linq.JObject containing query results and metadata defining the cached dataset.</returns>
+        public async Task<JObject> QueryDatasetAsync(string datasetName, string indexBy, string timeframe)
+        {
+            return await Datasets.Results(datasetName, indexBy, timeframe);
+        }
+
+        /// <summary>
+        /// Get query results from a Cached Dataset
+        /// </summary>
+        /// <param name="datasetName">Name of cached dataset to query.</param>
+        /// <param name="indexBy">The string property value you want to retrieve results by.</param>
+        /// <param name="timeframe">Limits retrieval of results to a specific portion of the Cached Dataset</param>
+        /// <returns>An instance of Newtonsoft.Json.Linq.JObject containing query results and metadata defining the cached dataset.</returns>
+        public JObject QueryDataset(string datasetName, string indexBy, string timeframe)
```

Comment:
Maybe the datasets return type could at least have `Result` that might be an `IEnumerable<>` if it's grouped/interval, or some other new `QueryResult` type we could introduce across the board to not just return `string` for all the metric queries. Then it could use `[JsonExtensionData]` to pack up the remaining properties, since there's some metadata and such that might be there.

Reply:
This is sort of related to Issue #56, but might require a separate issue apart from that.